File: LeetCode-Python/JumpGameII.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return 0

        jumps = 0
        current_max_reach = 0
        max_reach = 0

        for i in range(n - 1):
            max_reach = max(max_reach, i + nums[i])

            if i == current_max_reach:
                jumps += 1
                current_max_reach = max_reach

        return jumps

File: LeetCode-Cpp/JumpGameII.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        if (n == 1) {
            return 0;
        }

        int jumps = 0; // Atlama sayısı, sonuca ulaşmak için gereken minimum atlama sayısını temsil eder.
        int current_max_reach = 0; // Şu anki adımda ulaşabileceğiniz maksimum endeks.
        int max_reach = 0; // Daha fazla atlama yaparak ulaşabileceğiniz maksimum endeks.

        for (int i = 0; i < n - 1; i++) {
            max_reach = std::max(max_reach, i + nums[i]); // Şu anki adımda ulaşabileceğiniz maksimum endeksi günceller.

            if (i == current_max_reach) {
                jumps++;
                current_max_reach = max_reach; // Şu anki adımda ulaşabileceğiniz maksimum endeksi günceller ve bir atlama ekler.
            }
        }

        return jumps; 
        }
};